app/controllers/api/v1/rents_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Api
+  module V1
+    class RentsController < ApiV1Controller
+      def index
+        render_paginated Rent.rented_by(user_id).includes(:user, :book)
+      end
+
+      def create
+        rent = Rent.new(rent_params)
+        if rent.save
+          render json: rent, status: :created
+        else
+          render json: { errors: rent.errors.full_messages }, status: :unprocessable_entity
+        end
+      end
+
+      private
+
+      def rent_params
+        params.require(:rent).permit(:user_id, :book_id, :start_date, :end_date)
```

Comment:
Here you should require thess field, to avoid DB to work for nothing.

Reply:
I required these fields but the require method does not return a hash so I have to permit these fields later in order to be able to assign those fields to the rent model. Is there a better way to do that?

Reply:
Yes you did right. I never found other way to to this. If you have a better porposition or idea in the future do not be shy and share it :)